content/docs/rfcs/53/README.md

```diff
@@ -0,0 +1,316 @@
+---
+slug: 55
+title: 55/WAKU2-ORGANIZATION-CHATS
+name: Waku v2 Organization Chats
+status: raw
+category: Standards Track
+tags: waku-application
+editor: Aaryamann Challani <aaryamann@status.im>
+contributors:
+---
+
+# Abstract
+
+This document describes the design of organization chats for Waku v2, allowing for multiple users to communicate in a group chat. 
+This is a key feature for the Status messaging app. 
+
+# Background and Motivation
+
+Large group chats enable organizations to communicate.
+This would require channels, which are subject-based.
+The messages in a channel are broadcasted to all the users in the channel.
+
+A regular group chat between two or more peers reduces to a 1:1 chat between each peer and the other peers.
+One mechanism for 1:1 chats is described in [53/WAKU2-X3DH](https://rfc.vac.dev/spec/53/).
+However, this method does not scale as the number of peers increases, for the following reasons -
+1. The number of messages sent over the network increases as the number of peers increases.
+2. Handling the X3DH key exchange for each peer is computationally expensive.
+
+Having multicast channels reduces the overhead of a regular group chat.
+Additionally, if all the peers have a shared key, then the number of messages sent over the network is reduced to one per message.
+
+# Design Requirements
+
+Due to the nature of organizations, the following requirements are necessary for the design of organization chats -
+
+1. The Organization owner is trusted.
+2. The Organization owner can add or remove peers from the Organization.
+3. The Organization owner can add or remove channels.
+4. The peers in the Organization can send/receive messages to the channels which they have access to.
+5. Organizations may be encrypted or unencrypted (public).
+6. An Organization is uniquely identified by a public key.
+7. The public key of the Organization is shared out of band.
+8. The metadata of the Organization can be found by listening on a content topic derived from the public key of the Organization.
+9. Peers part of an Organization run their own infrastructure.
```

Comment:
@felicio.eth can you confirm if these assumptions are OK?

Reply:
LGTM, just,

> 1. The Organization owner is trusted.

I'd replace org for community.

> 2. The Organization owner can add or remove peers from the Organization.

Specifically, those are kick and ban ops (ref https://discord.com/channels/624205794384281629/730410574428504166/1020292913768443985).

> 3. The Organization owner can add or remove channels.

And edit ofc, but dunno how taxative we wanna be here.

> 5. Organizations may be encrypted or unencrypted (public).

Might be better to primarily distinguish them as private and public (ref https://discord.com/channels/624205794384281629/1048212980904501329/1052603998714220564).

Also, and maybe just as an fyi, public communities are "de facto unencrypted" atm. Since they're encrypted with a symmetric key, which is however obtainable by anyone (ref. https://discord.com/channels/864066763682218004/865466680129880074/1058202405583474688). If you'd be questioning this, I'd probably agree with you.

> 9. Peers part of an Organization run their own infrastructure.
 
Meaning running Store or Relay protocols, for example? Web clients don't do that for instance.

Reply:
Also, admin role def at https://discord.com/channels/624205794384281629/697843216048717905/1022799536495796224.

Reply:
> 4. The peers in the Organization can send/receive messages to the channels which they have access to.

Members may be more suitable naming.

Reply:
Given op thread at https://discord.com/channels/864066763682218004/1064649971497971862/1065146990540902400, do you think the following is worth mentioning here too:

- the periodic publishing
- the clock
- the token-gating
- waku protocols in use
- back-ups
- resending (not familiar with it myself yet)
- notifications

Reply:
Thanks for your feedback.
There are a couple things we need to discuss though -
1. Since status communities "implements" this spec, the implementation details of web clients will be omitted, as this only serves as a way on *how* communities work.
2. unencrypted vs encrypted communities: as per my understanding of the message you shared, it is possible to derive the key of public communities. Is there a reason we're encrypting messages from public communities at all?
3. Based on naming in other specs, let's try to stick to "peers" instead of "members". No strong opinion here, but can change it if it's required.
4. I agree that we need to mention periodic publishing, clock, waku protocols in use, and backups. I believe token gating is still being workshopped, so a new PR to this rfc can discuss token gating and relevant ideas/issues there. I'll add it to future work for now.
5. Imo we don't need to *specify* how notifications are delivered. 

Addressed some of your concerns in [6352e9f](https://github.com/vacp2p/rfc/pull/567/commits/6352e9fb6602ba8860ea161ae840803ba5a5594b)

Reply:
Roger that.

> Addressed some of your concerns in [6352e9f](https://github.com/vacp2p/rfc/pull/567/commits/6352e9fb6602ba8860ea161ae840803ba5a5594b)

Nice. Would you agree to move "To store and retrieve messages for offline devices" line to "SHOULD" section too instead of "MAY"? Without it, users wouldn't be able to fetch historical messages, for example.

> Is there a reason we're encrypting messages from public communities at all?

I don't have a better answer, I'm sorry. @fryorcraken, please do you?

Reply:
Done, moved WAKU2-STORE to the "SHOULD" section

Reply:
> > Is there a reason we're encrypting messages from public communities at all?
> 
> I don't have a better answer, I'm sorry. @fryorcraken, please do you?

Sorry, I don't know either, would need to refer to @cammellos

Reply:
Historical reasons, it was always a requirement for waku-1/whisper that messages have to be encrypted when sent over the network (there used to be support for unencrypted but it was dropped).
One benefit is that it provides some level of obscurity, since if you don't know/share the public key of the community, traffic can't be encrypted, but it's very bland and that's why we consider them unencrypted.